### PR TITLE
Remove deprecated class ArrayCollection

### PR DESCRIPTION
**Pull request by mroest**
In preparation for a SF4 upgrade this removes the deprecated ArrayCollection
class.

### CODE REVIEW

**Review by MKodde — CHANGES_REQUESTED** [review]
The change looks good. Some suggestions:

1. Add type definitions, now that we are no longer supporting PHP 5.4, this is a good way to make the code more reliable.
2. Consider adding a simple unit test to verify the correct working of this repo.

Other than that, nice to see the removal of the dependency on the array collection!

File: src/Surfnet/StepupGateway/SamlStepupProviderBundle/Provider/ProviderRepository.php [hunk 1]
```diff
@@ -28,28 +27,28 @@
 final class ProviderRepository
 {
     /**
-     * @var <ArrayCollection>Provider
+     * @var []]Provider
```
Comment (thijskh) [inline src/Surfnet/StepupGateway/SamlStepupProviderBundle/Provider/ProviderRepository.php:30]: typo here?

File: src/Surfnet/StepupGateway/SamlStepupProviderBundle/Tests/Provider/ProviderRepositoryTest.php [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+use PHPUnit\Framework\TestCase;
+use Surfnet\SamlBundle\Entity\IdentityProvider;
+use Surfnet\SamlBundle\Entity\ServiceProvider;
+use Surfnet\StepupGateway\SamlStepupProviderBundle\Exception\InvalidConfigurationException;
+use Surfnet\StepupGateway\SamlStepupProviderBundle\Exception\UnknownProviderException;
+use Surfnet\StepupGateway\SamlStepupProviderBundle\Provider\Provider;
+use Surfnet\StepupGateway\SamlStepupProviderBundle\Provider\ProviderRepository;
+use Surfnet\StepupGateway\SamlStepupProviderBundle\Saml\StateHandler;
+
+class ProviderRepositoryTest extends TestCase
```
Comment (MKodde) [inline src/Surfnet/StepupGateway/SamlStepupProviderBundle/Tests/Provider/ProviderRepositoryTest.php:30]: :muscle:

**Review by MKodde — APPROVED** [review]
Except for the typo, LGTM. Thanks for adding the test!

File: src/Surfnet/StepupGateway/SamlStepupProviderBundle/Provider/ProviderRepository.php (hunk 1 above)
Comment (mroest) [reply src/Surfnet/StepupGateway/SamlStepupProviderBundle/Provider/ProviderRepository.php]: Thanks! Good Catch! Fixed

**Pull request merged**